### ISSUE TITLE
Inconsistency in how range bounds are handled

### ISSUE THREAD

**Issue by hmaurer** [date-fns/date-fns#901]
I stumbled upon a pretty major inconsistency tonight in how range bounds are treating in the library. First of all, I was a little surprised to find that `isWithinRange` treats both bounds as inclusive, which makes it a little difficult to handle some scenarios. I won't go into detail on this as it has already been discussed in #366. 

The bigger issue is that it seems that the `areRangesOverlapping` function does *not* treat the end-bound as inclusive. I'll illustrate with an example:

```
dateFns.areRangesOverlapping(
  new Date(2014, 0, 10), new Date(2014, 0, 17), new Date(2014, 0, 17), new Date(2014, 0, 21)
)
// => false

isWithinRange(new Date(2014, 0, 17), new Date(2014, 0, 10), new Date(2014, 0, 17))
// => true
```

In other words, given those two ranges, we are in a situation where according to `areRangesOverlapping` they do not overlap, but according to `isWithinRange` the endpoint of the second range is within the first range; an obvious contradiction.

I personally lean on the side of treating all ranges as `[)` (start-bound inclusive, end-bound exclusive) because it gives the ability to easily express consecutive intervals (i.e. from 12pm until 1pm, then from 1pm until 3pm, etc), a requirement in my application. That said, I think regardless of how range endpoint are treated it's important to have consistency across all functions dealing with ranges in order to avoid confusing scenarios like the one I just outlined :-)
